test/e2e/volume-provisioner/framework/framework.go

```diff
@@ -315,3 +302,373 @@ func (f *Framework) newConfigurationProvider(cfg *core.Config) (common.Configura
+					Containers: []v1.Container{{
+						Name:    "oci-fexvolume-driver",
+						Image:   TestContext.Image,
+						Command: []string{"/bin/bash", "/usr/local/bin/install.sh"},
+						SecurityContext: &v1.SecurityContext{
+							Privileged: utilpointer.BoolPtr(true),
+						},
+						VolumeMounts: []v1.VolumeMount{{
+							Name:      "flexvolume-mount",
+							MountPath: "/flexmnt",
+						}, {
+							Name:      "config",
+							MountPath: "/tmp",
+							ReadOnly:  true,
+						}, {
+							Name:      "kubeconfig",
+							MountPath: "/tmp2",
+							ReadOnly:  true,
+						}},
+					}},
+					Volumes: []v1.Volume{{
+						Name: "config",
+						VolumeSource: v1.VolumeSource{
+							Secret: &v1.SecretVolumeSource{
+								SecretName: "oci-flexvolume-driver",
+							},
+						},
+					}, {
+						Name: "kubeconfig",
+						VolumeSource: v1.VolumeSource{
+							Secret: &v1.SecretVolumeSource{
+								SecretName: "oci-flexvolume-driver-kubeconfig",
+							},
+						},
+					}, {
+						Name: "flexvolume-mount",
+						VolumeSource: v1.VolumeSource{
+							HostPath: &v1.HostPathVolumeSource{
+								Path: "/usr/libexec/kubernetes/kubelet-plugins/volume/exec/",
+								Type: &hostPathDirectoryOrCreate,
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+
+	workerDS := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "oci-flexvolume-driver-worker",
+			Namespace: "kube-system",
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": "oci-flexvolume-driver-worker"},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "oci-flexvolume-driver-worker",
+					Labels: map[string]string{"app": "oci-flexvolume-driver-worker"},
+				},
+				Spec: v1.PodSpec{
+					ServiceAccountName: "oci-flexvolume-driver",
+					Containers: []v1.Container{{
+						Name:    "oci-fexvolume-driver",
+						Image:   TestContext.Image,
+						Command: []string{"/bin/bash", "/usr/local/bin/install.sh"},
+						SecurityContext: &v1.SecurityContext{
+							Privileged: utilpointer.BoolPtr(true),
+						},
+						VolumeMounts: []v1.VolumeMount{{
+							Name:      "flexvolume-mount",
+							MountPath: "/flexmnt",
+						}},
+					}},
+					Volumes: []v1.Volume{{
+						Name: "flexvolume-mount",
+						VolumeSource: v1.VolumeSource{
+							HostPath: &v1.HostPathVolumeSource{
+								Path: "/usr/libexec/kubernetes/kubelet-plugins/volume/exec/",
+								Type: &hostPathDirectoryOrCreate,
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+
+	if err := createAndAwaitDaemonSet(client, masterDS); err != nil {
+		return errors.Wrap(err, "installing flexvolume driver master DaemonSet")
+	}
+
+	if err := createAndAwaitDaemonSet(client, workerDS); err != nil {
+		return errors.Wrap(err, "installing flexvolume driver worker DaemonSet")
+	}
+
+	return nil
+}
+
+func createAndAwaitDaemonSet(client clientset.Interface, desired *appsv1.DaemonSet) error {
```

Comment:
Can we move this further up in the framework? It's not specific to the flex volume driver.